### PR TITLE
Add Qute TextMate grammar and language configuration for JSON, YAML and txt files

### LINKED ISSUE #182: Add Qute TextMate grammar and language configuration for HTML files
Create a `qute.tmLanguage` which includes html TextMate like https://github.com/angelozerr/lsp4e-freemarker/blob/master/org.eclipse.lsp4e.freemarker/syntaxes/ftl.tmLanguage
You can see a demo in this repository with HTML completion (coming from the HTML Language Server of vscode and Freemarker Language Server for validation).

Perhaps a good idea is to use those file extension:

*.html.qute (or if it doesn't work .htmlQute)
*.json.qute (or if it doesn't work .jsonQute)

This language configuration could be used in Eclipse IDE too with TM4E  /cc @jeffmaury  (and I think IJ supports too TextMate)

### PR DESCRIPTION
**Pull request by xorye**
fixes #182 

I will write some of my concerns about this PR tomorrow (mostly about how commenting should work).

For the time being, here are some quick demos:

*.qute.json:
image

*.qute.yaml:
image

*.qute.txt:
image

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,6 +73,36 @@
...
+        ],
+        "extensions": [
+          "qute.yaml"
+        ],
+        "configuration": "./language-support/qute/language-configuration.json"
+      },
+      {
+        "id": "qute-txt",
+        "aliases": [
+          "Qute txt"
```
Comment (fbricon) [inline package.json]: Qute Text

**Review by fbricon — CHANGES_REQUESTED** [review]

**Comment by fbricon** [comment]
@xorye LGTM, except for the small nitpick about `Qute txt`. What's the issue with comments?

**Comment by xorye** [comment]
Hi @maxandersen 

In `*.qute.json` files, do you know if comments are valid? If they are, do you know if the comments should be in this format`{! !}`?

And for `*.qute.yaml` files, should they be commented with `{! !}`?

**Comment by fbricon** [comment]
Given:
```
{
    // regular json shouldn't support that kind of comment
    "foo":"bar"
    "bar": "this is {name} speaking"
    {! qute-style comments should work !}
}
```
This renders as:
Screen Shot 2020-01-20 at 11 06 25 AM

**Comment by xorye** [comment]
@fbricon fixed
image

But for the regular json comment, the default json grammar does in fact highlight those comments:
image

It looks like VSCode offers validation for .json files to notify the user that comments are not valid

File: language-support/qute/qute-txt.tmLanguage.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Qute txt",
```
Comment (fbricon) [inline language-support/qute/qute-txt.tmLanguage.json]: Qute Text

File: language-support/qute/qute-json.tmLanguage.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      ]
+    }
+  },
+  "patterns": [
+    {
+      "include": "source.json"
+    }
+  ],
+  "scopeName": "source.json.qute",
+  "uuid": "C011BB07-875D-4DEB-9582-0E1FEC0D1E6E"
```
Comment (fbricon) [inline language-support/qute/qute-json.tmLanguage.json]: see https://stackoverflow.com/a/59791552/753170

Comment (xorye) [reply language-support/qute/qute-json.tmLanguage.json]: yes, that was me who asked that :)

**Comment by fbricon** [comment]
qute text is broken:
```
this is {name} speaking
{
    // regular text shouldn't crap out like that
    wat this is {that}?
    {! qute-style comments should work !}
}
```
it doesn't like the brackets
Screen Shot 2020-01-20 at 5 33 00 PM

**Comment by xorye** [comment]
After testing out Qute with the Quarkus Quickstarts sample project: https://github.com/quarkusio/quarkus-quickstarts/tree/master/qute-quickstart
it looks like there cannot be a space or newline when using `{` and `}`, 

ie, `{item.price}` is valid, `{ item.price }`, `{ item.price}` and `{item.price }` are not. Same goes for `{#if` and `{#for`,

I pushed changes for the Qute grammar that takes that into account. 

Demo:
image

**Comment by xorye** [comment]
I found a new problem: 
image

Fixing it now

**Comment by xorye** [comment]
^ That problem is a bit tougher to solve.

It would be helpful if documentation for escaping `{` and `}` was finalized: https://github.com/quarkusio/quarkus/issues/6621

Handlebars suffers from a similar problem:
image

The opening `{{` makes it believe that there will be an expression and a closing `}}` that follows it, but since there is no closing `}}`, the grammar thinks that everything until the end of the file is the expression.

If the user truly wanted to have `{{hello` in their HTML, it would be better for the grammar if `&#123;` or `\{` was used instead of `{`.

**Comment by fbricon** [comment]
@xorye please open a separate ticket for the above issue

**Pull request merged**